### PR TITLE
fix AC warning method redefined.

### PR DESCRIPTION
**Pull request by utilum**
### Fixes the following warning

```
[oz@localhost actioncable]$ bundle exec rake test
/home/oz/.rbenv/versions/2.3.3/bin/ruby -w -I"lib:test" -I"/home/oz/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-11.3.0/lib" "/home/oz/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-11.3.0/lib/rake/rake_test_loader.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/inline_test.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/async_test.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/base_test.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/postgresql_test.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/evented_redis_test.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/redis_test.rb" "/home/oz/code/rails/actioncable/test/subscription_adapter/subscriber_map_test.rb" "/home/oz/code/rails/actioncable/test/worker_test.rb" "/home/oz/code/rails/actioncable/test/connection/identifier_test.rb" "/home/oz/code/rails/actioncable/test/connection/stream_test.rb" "/home/oz/code/rails/actioncable/test/connection/subscriptions_test.rb" "/home/oz/code/rails/actioncable/test/connection/base_test.rb" "/home/oz/code/rails/actioncable/test/connection/client_socket_test.rb" "/home/oz/code/rails/actioncable/test/connection/multiple_identifiers_test.rb" "/home/oz/code/rails/actioncable/test/connection/cross_site_forgery_test.rb" "/home/oz/code/rails/actioncable/test/connection/string_identifier_test.rb" "/home/oz/code/rails/actioncable/test/connection/authorization_test.rb" "/home/oz/code/rails/actioncable/test/client_test.rb" "/home/oz/code/rails/actioncable/test/channel/rejection_test.rb" "/home/oz/code/rails/actioncable/test/channel/stream_test.rb" "/home/oz/code/rails/actioncable/test/channel/base_test.rb" "/home/oz/code/rails/actioncable/test/channel/broadcasting_test.rb" "/home/oz/code/rails/actioncable/test/channel/naming_test.rb" "/home/oz/code/rails/actioncable/test/channel/periodic_timers_test.rb" "/home/oz/code/rails/actioncable/test/server/base_test.rb" "/home/oz/code/rails/actioncable/test/server/broadcasting_test.rb" 
Run options: --seed 41014

# Running:

....................................................................................../home/oz/code/rails/actioncable/lib/action_cable/remote_connections.rb:48: warning: method redefined; discarding old identifiers
/home/oz/code/rails/activesupport/lib/active_support/core_ext/class/attribute.rb:110: warning: previous definition of identifiers was here
..................................................

```

ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux]
Rails master

### CODE REVIEW

File: actioncable/lib/action_cable/remote_connections.rb [hunk 1]
```diff
@@ -45,6 +45,7 @@ def disconnect
         end
 
         # Returns all the identifiers that were applied to this connection.
+        remove_method :identifiers
```
Comment (rafaelfranca) [inline actioncable/lib/action_cable/remote_connections.rb]: It doesn't make sense to remove a method that we defined ourselves. Can we just change the `class_attribute` definition to not define the instance method?

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: actioncable/lib/action_cable/remote_connections.rb (hunk 1 above)
Comment (utilum) [inline actioncable/lib/action_cable/remote_connections.rb]: Thank you. It does seem strange, and I'm sure we can, but I'm getting lost in the code. You don't just mean renaming it, do you?

**Comment by utilum** [comment]
Not defining the class attribute raised errors, but eliminating the method has the same effect and raises naught. Looks like it is not necessary once `class_attribute` defines it.

**Comment by rafaelfranca** [comment]
We do need this method. Otherwise it will read the class attribute of the `RemoteConnection` class not the `connection_identifiers` of the server object.

**Comment by utilum** [comment]
OK, I see that. However, the method defined by `class_attribute` is used all over the place... am I right to think it's either a largish refactor or let the warnings warn?

**Comment by utilum** [comment]
Also, since we do need the instance method, don't we need a test for it? Is it OK for the suite to run smoothly if the method is deleted?

**Comment by utilum** [comment]
Can no longer replicate. Must have been cleaned up.

**Pull request closed without merging**